1_fetch/src/get_nwis_data.R

```diff
@@ -1,39 +1,35 @@
-download_nwis_data <- function(site_nums = c("01427207", "01432160", "01436690", "01466500"), Save_as_csv = FALSE){
-
-  parameterCd <-  '00010'
-  startDate <-"2014-05-01"
-  endDate <- "2015-05-01"
+download_nwis_data <- function(site_num, parameterCd = '00010', startDate = "2014-05-01", endDate = "2015-05-01"){
   
-  data_out <- data.frame()
+  data_out <- readNWISdata(sites = site_num, parameterCd = parameterCd, startDate = startDate, endDate = endDate, service="iv")
   
-  for(site_num in site_nums){
-
-    data <- readNWISdata(sites=site_num, service="iv", parameterCd = parameterCd, startDate = startDate, endDate = endDate)
-
-    ## COMMENTED THIS OUT as it breaks the code, tbd why    
-    
-    # -- simulating a failure-prone web-service here, do not edit --
-    set.seed(Sys.time())
-    if (sample(c(T,F,F,F), 1)){
-      stop(site_num, ' has failed due to connection timeout. Try tar_make() again')
-    }
+  # -- simulating a failure-prone web-service here, do not edit --
+  set.seed(Sys.time())
+  if (sample(c(T,F,F,F), 1)){
+    stop(site_num, ' has failed due to connection timeout. Try tar_make() again')
+  }
```

Comment:
Still have this in here. So indeed, sometime by nwis data downloads fail, but I can see the error in the pipeline comments right after the specific nwis site number is printed, so I know right away what interation is causing issues and can go in and debug there if needed.